client/components/status/status.js

```diff
@@ -87,65 +90,84 @@ export default class Status extends React.Component {
+      }
+      this.setState({username, password, userInfo}, () => {
+        // if the user is being automatically logged in but it's not
+        // active anymore (eg: has been banned)
+        // automatically perform log out
+        if (is_active === false) {
+          this.handleLogout(false);
         }
-        this.setState({username, password, userInfo}, () => {
-          // if the user is being automatically logged in but it's not
-          // active anymore (eg: has been banned)
-          // automatically perform log out
-          if (is_active === false) {
-            this.handleLogout(false);
-          }
-        });
-        if (isValid && is_active) {
-          const macaddr = cookies.get(`${orgSlug}_macaddr`);
-
-          if (macaddr) {
-            const params = {macaddr};
-            await this.getUserActiveRadiusSessions(params);
-            /* request to captive portal is made only if there is
-              no active session from macaddr stored in the cookie */
-            const {activeSessions} = this.state;
-            if (activeSessions && activeSessions.length === 0) {
-              if (this.loginFormRef && this.loginFormRef.current)
-                this.loginFormRef.current.submit();
-            }
-          } else if (
-            this.loginFormRef &&
-            this.loginFormRef.current &&
-            justAuthenticated
-          ) {
+      });
+
+      // stop here if user is banned
+      if (is_active === false) {
+        return;
+      }
+
+      const macaddr = cookies.get(`${orgSlug}_macaddr`);
+      if (macaddr) {
+        const params = {macaddr};
+        await this.getUserActiveRadiusSessions(params);
+        /* request to captive portal is made only if there is
+          no active session from macaddr stored in the cookie */
+        const {activeSessions} = this.state;
+        if (activeSessions && activeSessions.length === 0) {
+          if (this.loginFormRef && this.loginFormRef.current)
             this.loginFormRef.current.submit();
-            setUserData({...userData, justAuthenticated: false});
-          }
-          await this.getUserActiveRadiusSessions();
-          await this.getUserPassedRadiusSessions();
-          const intervalId = setInterval(() => {
-            this.getUserActiveRadiusSessions();
-          }, 60000);
-          this.setState({intervalId});
-          window.addEventListener("resize", this.updateScreenWidth);
-          this.updateSpinner();
         }
+      } else if (
+        this.loginFormRef &&
+        this.loginFormRef.current &&
+        justAuthenticated
+      ) {
+        this.loginFormRef.current.submit();
+        userData.justAuthenticated = false;
+        setUserData(userData);
+      }
+
+      if (needsVerify("bank_card", userData, settings)) {
+        window.location.assign(userData.payment_url);
+      }
+
+      if (userData.is_verified) {
+        setLoading(false);
+      } else {
+        return;
```

Comment:
Instead of return why don't we logout the user?

Reply:
because the user needs to do the verification process and needs to be logged in to do that

Reply:
Yeah, I understood.